std/variant.d

```diff
@@ -249,11 +249,21 @@ private:
         {
             static if (is(typeof(*rhsPA == *zis)))
             {
-                if (*rhsPA == *zis)
+                enum isEmptyStructWithoutOpEquals = (is(A == struct) && A.tupleof.length == 0 &&
+                                                     !__traits(hasMember, A, "opEquals"));
+                static if (!isEmptyStructWithoutOpEquals)
+                {
+                    if (*rhsPA == *zis)
+                        return 0;
+                }
+                static if (isEmptyStructWithoutOpEquals)
                 {
                     return 0;
+                    // The check above will always succeed if A is an empty struct.
```

Comment:
Would the comment usually go before the code?

Reply:
Fixed.